src/components/ShareTrajectoryModal/EmbedSnippetPanel.tsx

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import { Input, Checkbox, Button } from "antd";
+import { URL_PARAM_KEY_FILE_NAME } from "../../constants";
+import VerticalFlexbox from "../../styles/utils";
+import { getUrlParamValue, copyToClipboard } from "../../util";
+import { DownArrow } from "../Icons";
+
+import styles from "./style.css";
+
+const DEFAULT_WIDTH = 834;
+const DEFAULT_HEIGHT = 480;
+const MAX_WIDTH = 3840;
+const MAX_HEIGHT = 2160;
+
+interface EmbedSnippetPanelProps {
+    startTime: number;
+}
+
+const EmbedSnippetPanel = ({ startTime }: EmbedSnippetPanelProps) => {
+    const trajectory = getUrlParamValue(
+        window.location.href,
+        URL_PARAM_KEY_FILE_NAME
+    );
+
+    const [width, setWidth] = React.useState(DEFAULT_WIDTH);
+    const [height, setHeight] = React.useState(DEFAULT_HEIGHT);
+    const [constrainProportions, setConstrainProportions] =
+        React.useState(false);
+
+    const [showEmbedSettingsPanel, setShowEmbedSettingsPanel] =
+        React.useState(false);
+
+    const url = `https://simularium.allencell.org/embed?trajFileName=${trajectory}&t=${startTime}`;
+    const embedSnippet = `<iframe height="${width}" width="${height}" src="${url}" title="Simularium" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`;
+    const inputIsValid = (input: string) => {
+        return !Number.isNaN(parseInt(input));
+    };
+
+    const handleChangeWidth = (input: string) => {
+        const currentRatio = width / height;
+        const numericInput = inputIsValid(input) ? parseInt(input) : width;
+        const roundedLimitedInput = Math.min(MAX_WIDTH, numericInput);
+        setWidth(roundedLimitedInput);
+        if (constrainProportions) {
+            setHeight(Math.round(roundedLimitedInput / currentRatio));
+        }
+    };
+
+    const handleChangeHeight = (input: string) => {
```

Comment:
these two functions have a lot of the same code, but you don't gain a lot by combining them because you still need numerous if/else statements (for the max value, and for setting the opposite value), which makes things less clear. In my opinion, the two functions is easier to read than 1 with multiple if elses